### PR TITLE
Check if indexPath is valid when accessing chat buttons

### PR DESCRIPTION
**Pull request by yuliaveres**
Fixes invalid indexPath crash
https://fabric.io/bakken--bck2/ios/apps/com.bakkenbaeck.token/issues/59516ed8be077a4dcca7033d?time=last-twenty-four-hours

### CODE REVIEW

**Comment by yuliaveres** [comment]
@roberthein , i didn't but in general it is good practise to always check indexPath 😉

File: Toshi/Extensions/ControlsCollectionViewDelegateDatasource.swift [hunk 1]
```diff
@@ -59,10 +59,10 @@ class SubcontrolsViewDelegateDatasource: NSObject, UICollectionViewDataSource, U
     }
 
     func didTapButton(for cell: ControlCell) {
-        guard let indexPath = self.subcontrolsCollectionView?.indexPath(for: cell) else { return }
+        guard let indexPath = self.subcontrolsCollectionView?.indexPath(for: cell), indexPath.item != NSNotFound, indexPath.item < self.items.count else { return }
```
Comment (Elland) [inline Toshi/Extensions/ControlsCollectionViewDelegateDatasource.swift]: Should probably abstract this. Extension method?

Comment (yuliaveres) [inline Toshi/Extensions/ControlsCollectionViewDelegateDatasource.swift]: Extension for indexPath ?

Comment (Elland) [inline Toshi/Extensions/ControlsCollectionViewDelegateDatasource.swift]: To check for validity. `func isValid(for count: Int) -> Bool` or something.

Comment (Elland) [inline Toshi/Extensions/ControlsCollectionViewDelegateDatasource.swift]: What do you think? Since it's used in more than one place.

Comment (yuliaveres) [inline Toshi/Extensions/ControlsCollectionViewDelegateDatasource.swift]: yeah, 🤔  good idea i think

**Pull request merged**